src/Task.cc

```diff
@@ -2754,12 +2754,14 @@ static void set_up_process(Session& session, const ScopedFd& err_fd,
     setsid();
   }
 
+#if defined(__i386__) || defined(__x86_64)
```

Comment:
I think I'd prefer you put this in an arch function.

Reply:
Ok